py/kubeflow/testing/run_e2e_workflow.py

```diff
@@ -124,9 +124,10 @@ def run(args, file_handler): # pylint: disable=too-many-statements,too-many-bran
   diff_command = []
   if job_type == "presubmit":
     # We need to get a common ancestor for the PR and the base branch
-    common_ancestor = util.run(
-      ["git", "merge-base", "HEAD", base_branch_name],
-      cwd=os.path.join(args.repos_dir, repo_owner, repo_name))
+    cloned_repo_dir = os.path.join(args.repos_dir, repo_owner, repo_name)
+    _ = util.run(["git", "checkout", base_branch_name], cwd=cloned_repo_dir)
```

Comment:
Do you want to checkout base_branch_name or just fetch it?

If you check it out then I think the current version of the checked out code would be set to v0.5-branch which isn't what we want since we want the current branch to be the branch corresponding to the PR.

Reply:
Thanks for pointing this out! Yeah I should just do `git merge-base remotes/origin/blah` instead of a `git checkout` as this changes the HEAD.

Now I'm doing a `git fetch origin blah` to prevent the case where `remotes/origin/blah` may not exist locally (I can't think of a case though but just doing that for safety) and then `git merge-base remotes/origin/blah`.

PTAL?